---a/net/minecraft/entity/EntityMinecartCommandBlock.java
+++b/net/minecraft/entity/EntityMinecartCommandBlock.java
@@ -1,6 +1,5 @@
 package net.minecraft.entity;
 
-import io.netty.buffer.ByteBuf;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.command.server.CommandBlockLogic;
 import net.minecraft.entity.item.EntityMinecart;
@@ -11,129 +11,107 @@
 import net.minecraft.util.IChatComponent;
 import net.minecraft.util.Vec3;
 import net.minecraft.world.World;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
-public class EntityMinecartCommandBlock extends EntityMinecart
-{
-    private final CommandBlockLogic commandBlockLogic = new CommandBlockLogic()
-    {
-        private static final String __OBFID = "CL_00001673";
-        public void func_145756_e()
+public class EntityMinecartCommandBlock extends EntityMinecart {
+
+    // SpigotMCP Start
+    private final CommandBlockLogic commandBlockLogic = new EntityMinecartCommandBlockListener();
+
+    public class EntityMinecartCommandBlockListener extends CommandBlockLogic {
+
         {
+            this.sender = (org.bukkit.craftbukkit.entity.CraftMinecartCommand) EntityMinecartCommandBlock.this.getBukkitEntity(); // CraftBukkit - Set the sender
+        }
+
+        private static final String __OBFID = "CL_00001673";
+
+        public void func_145756_e() {
             EntityMinecartCommandBlock.this.getDataWatcher().updateObject(23, this.getCustomName());
             EntityMinecartCommandBlock.this.getDataWatcher().updateObject(24, IChatComponent.Serializer.componentToJson(this.getLastOutput()));
         }
-        @SideOnly(Side.CLIENT)
-        public int func_145751_f()
-        {
-            return 1;
-        }
-        @SideOnly(Side.CLIENT)
-        public void func_145757_a(ByteBuf p_145757_1_)
-        {
-            p_145757_1_.writeInt(EntityMinecartCommandBlock.this.getEntityId());
-        }
-        public BlockPos getPosition()
-        {
+
+        public BlockPos getPosition() {
             return new BlockPos(EntityMinecartCommandBlock.this.posX, EntityMinecartCommandBlock.this.posY + 0.5D, EntityMinecartCommandBlock.this.posZ);
         }
-        public Vec3 getPositionVector()
-        {
+
+        public Vec3 getPositionVector() {
             return new Vec3(EntityMinecartCommandBlock.this.posX, EntityMinecartCommandBlock.this.posY, EntityMinecartCommandBlock.this.posZ);
         }
-        public World getEntityWorld()
-        {
+
+        public World getEntityWorld() {
             return EntityMinecartCommandBlock.this.worldObj;
         }
-        public Entity getCommandSenderEntity()
-        {
+
+        public Entity getCommandSenderEntity() {
             return EntityMinecartCommandBlock.this;
         }
-    };
+    }
+
+    ;
+    // SpigotMCP end
     private int activatorRailCooldown = 0;
     private static final String __OBFID = "CL_00001672";
 
-    public EntityMinecartCommandBlock(World worldIn)
-    {
+    public EntityMinecartCommandBlock(World worldIn) {
         super(worldIn);
     }
 
-    public EntityMinecartCommandBlock(World worldIn, double p_i45322_2_, double p_i45322_4_, double p_i45322_6_)
-    {
+    public EntityMinecartCommandBlock(World worldIn, double p_i45322_2_, double p_i45322_4_, double p_i45322_6_) {
         super(worldIn, p_i45322_2_, p_i45322_4_, p_i45322_6_);
     }
 
-    protected void entityInit()
-    {
+    protected void entityInit() {
         super.entityInit();
         this.getDataWatcher().addObject(23, "");
         this.getDataWatcher().addObject(24, "");
     }
 
-    protected void readEntityFromNBT(NBTTagCompound tagCompund)
-    {
+    protected void readEntityFromNBT(NBTTagCompound tagCompund) {
         super.readEntityFromNBT(tagCompund);
         this.commandBlockLogic.readDataFromNBT(tagCompund);
         this.getDataWatcher().updateObject(23, this.getCommandBlockLogic().getCustomName());
         this.getDataWatcher().updateObject(24, IChatComponent.Serializer.componentToJson(this.getCommandBlockLogic().getLastOutput()));
     }
 
-    protected void writeEntityToNBT(NBTTagCompound tagCompound)
-    {
+    protected void writeEntityToNBT(NBTTagCompound tagCompound) {
         super.writeEntityToNBT(tagCompound);
         this.commandBlockLogic.writeDataToNBT(tagCompound);
     }
 
-    public EntityMinecart.EnumMinecartType getMinecartType()
-    {
+    public EntityMinecart.EnumMinecartType getMinecartType() {
         return EntityMinecart.EnumMinecartType.COMMAND_BLOCK;
     }
 
-    public IBlockState getDefaultDisplayTile()
-    {
+    public IBlockState getDefaultDisplayTile() {
         return Blocks.command_block.getDefaultState();
     }
 
-    public CommandBlockLogic getCommandBlockLogic()
-    {
+    public CommandBlockLogic getCommandBlockLogic() {
         return this.commandBlockLogic;
     }
 
-    public void onActivatorRailPass(int p_96095_1_, int p_96095_2_, int p_96095_3_, boolean p_96095_4_)
-    {
-        if (p_96095_4_ && this.ticksExisted - this.activatorRailCooldown >= 4)
-        {
+    public void onActivatorRailPass(int p_96095_1_, int p_96095_2_, int p_96095_3_, boolean p_96095_4_) {
+        if (p_96095_4_ && this.ticksExisted - this.activatorRailCooldown >= 4) {
             this.getCommandBlockLogic().trigger(this.worldObj);
             this.activatorRailCooldown = this.ticksExisted;
         }
     }
 
-    public boolean interactFirst(EntityPlayer playerIn)
-    {
+    public boolean interactFirst(EntityPlayer playerIn) {
         this.commandBlockLogic.func_175574_a(playerIn);
         return false;
     }
 
-    public void func_145781_i(int p_145781_1_)
-    {
+    public void func_145781_i(int p_145781_1_) {
         super.func_145781_i(p_145781_1_);
 
-        if (p_145781_1_ == 24)
-        {
-            try
-            {
+        if (p_145781_1_ == 24) {
+            try {
                 this.commandBlockLogic.setLastOutput(IChatComponent.Serializer.jsonToComponent(this.getDataWatcher().getWatchableObjectString(24)));
-            }
-            catch (Throwable throwable)
-            {
+            } catch (Throwable throwable) {
                 ;
             }
-        }
-        else if (p_145781_1_ == 23)
-        {
+        } else if (p_145781_1_ == 23) {
             this.commandBlockLogic.setCommand(this.getDataWatcher().getWatchableObjectString(23));
         }
     }
-}
+}
